### DISCUSSION TITLE
export values on corresponding meshes (adaptive mesh refinement)

### DISCUSSION THREAD

**Discussion by maxxxzdn** [MPh-py/MPh#116] [General]
Hi John, 

I want to use Comsol to generate a lot of solutions of a time-dependent 1D PDE with different initial conditions and _adaptive mesh refinement_. 

Precisely, I want to export each time point of each solution with a corresponding adapted mesh. In Comsol, I can do it "simply" by creating a dataset for each solution and then downloading each time point manually (since meshes are different, I am not able to download all points at once). 

The approach obviously doesn't scale, which is a problem as I want to generate many solutions (up to 1k). Furthermore, I am looking for a way to automatize it. Can you please point out on a possible way to do it within the library?

Sincerely,
Maksim

**Comment by john-hen** [comment]
I am not familiar with your use case, so other than the general advice in #86, I got nothing.

**Reply by maxxxzdn** [reply]
I think the challenge for me is to take a computed COMSOL solution as .mph file and evaluate it on vertices of a specific mesh at each time point _iteratively_. I have all the meshes stored within the solution and the array of time points. 

something like values[t] = evaluate(model, mesh[t].vertices, t)

**Comment by maxxxzdn** [comment]
I just figured out that it also works if I export data for a specific time-point instead of all at once (see pic).
So my question practically boils down to whether I can do it _automatically_ in python.
<img width="720" alt="image" src="https://user-images.githubusercontent.com/50237680/218246128-1b0e5b16-7834-4822-8f55-a32e501dc8d4.png">

**Reply by john-hen** [reply]
I'm sure it's possible. You probably just have to figure out what parameters to pass to `model.evaluate()`. Like what `dataset` name and `inner` index. Which depends on how you've set up your model.

If that doesn't work, because of some detail I may be overlooking, then it's still possible to automate that in Python. Simply because you have access to the *full* Comsol API if need be. (Via MPh and thanks to JPype.) And virtually everything you do in the GUI can be automated, because that's how Comsol was designed.

MPh is just a Python wrapper with a number of convenience functions layered on top. It doesn't concern itself with most of Comsol's specifics.
